tests/unit/test_space.py

```diff
@@ -390,93 +398,178 @@ def _assert_correct_number_of_unique_constrained_samples(
     assert len(unique_samples) == len(samples)
 
 
-@pytest.mark.parametrize("num_samples", [0, 1, 10])
-def test_box_sampling_returns_correct_shape(num_samples: int) -> None:
-    box = Box(tf.zeros((3,)), tf.ones((3,)))
-    samples = box.sample(num_samples)
+def _box_sampling_constraints() -> Sequence[LinearConstraint]:
+    return [LinearConstraint(A=tf.eye(3), lb=tf.zeros((3)) + 0.3, ub=tf.ones((3)) - 0.3)]
+
+
+@pytest.mark.parametrize(
+    "num_samples, constraints",
+    ((n, c) for c in [None, _box_sampling_constraints()] for n in [0, 1, 10]),
+)
+def test_box_sampling_returns_correct_shape(
+    num_samples: int,
+    constraints: Sequence[LinearConstraint],
+) -> None:
+    box = Box(tf.zeros((3,)), tf.ones((3,)), constraints)
+    samples = box.sample_feasible(num_samples)
     _assert_correct_number_of_unique_constrained_samples(num_samples, box, samples)
 
 
-@pytest.mark.parametrize("num_samples", [0, 1, 10])
-def test_box_sobol_sampling_returns_correct_shape(num_samples: int) -> None:
-    box = Box(tf.zeros((3,)), tf.ones((3,)))
-    sobol_samples = box.sample_sobol(num_samples)
+@pytest.mark.parametrize(
```

Comment:
Probably more natural to write this as two parametrizes?

```suggestion
@pytest.mark.parametrize("num_samples", [0, 1, 1])
@pytest.mark.parametrize("constraints", [None, _box_sampling_constraints()])
```

Reply:
Done in b76379c.